pkg/gardenlet/controller/shoot/shoot_control.go

```diff
@@ -384,6 +401,28 @@ func (c *Controller) isSeedReadyForMigration(seed *gardencorev1beta1.Seed) error
 	return health.CheckSeedForMigration(seed, c.identity)
 }
 
+func (c *Controller) countShootBastions(ctx context.Context, shoot *gardencorev1beta1.Shoot, gardenClient kubernetes.Interface) (int, error) {
+	// list all bastions that reference this shoot
+	bastionList := operationsv1alpha1.BastionList{}
+	listOptions := client.ListOptions{
+		Namespace: shoot.Namespace,
+		// FieldSelector: fields.SelectorFromSet(fields.Set{operations.BastionShootName: shoot.Name}),
```

Comment:
Same here, this should also work, but doesn't :-D

Reply:
In order to make it work, you probably need something similar to https://github.com/gardener/gardener/blob/master/pkg/controllermanager/controller/factory.go#L201-L231

Reply:
Do you insist on using a Field selector? Peter and me had settled on just fetching all Bastions, considering how there will be probably very few Bastions at any given time in a project.

Reply:
I don't insist, probably your assumption is fair.